packages/amplify_datastore_plugin_interface/test/amplify_temporal_datetime_test.dart

```diff
@@ -134,16 +134,16 @@ void main() {
   });
 
   test('AWSDateTime from offset with single digit duration', () async {
-    DateTime now = DateTime.now();
+    DateTime dateTime = DateTime.parse("2021-11-09T18:53:12.183540Z");
```

Comment:
This is the test that I observed fail in CI. Initially I thought there might be a bug that was only causing failures for certain times, but I think this the code is working as intended and the test is the issue.

I was able to repeat that failure by testing with a DateTime with 0 milliseconds / microseconds. This fails the test because the format() method doesn't include milliseconds / microseconds if the value is 0, and the test is written in a way that assumes the string will include milliseconds / microseconds.

Here is code in the format() method that omits milliseconds / microseconds from the string if the value is 0. 
https://github.com/aws-amplify/amplify-flutter/blob/f658f9487ad94aface04ff9b0c361741ab3f9623/packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_datetime.dart#L144-L146

Here is an example of what the format() method will produce with and without milliseconds / microseconds
```dart
DateTime dateTime1 = DateTime.parse("2021-11-09T18:53:12.183540Z");
dateTime1.format(); // result is "2021-11-09T18:53:12.183540000Z"

DateTime dateTime2 = DateTime.parse("2021-11-09T18:53:12.000000Z");
dateTime2.format(); // result is "2021-11-09T18:53:12Z" **NOT** "2021-11-09T18:53:12.000000Z"
```

I am not sure if there is a specific reason to exclude ".000000", but it seems intentional and as far as I can tell isn't causing any actual issues with DS.